simulate/simulate.go

```diff
@@ -182,19 +182,25 @@ func MutateGene(inputSeq []dna.Base, branchLength float64, geneFile string) []dn
 				for codon := 0; codon < len(originalCodons); codon++ {
 					thisCodon := originalCodons[codon]
 					start = CheckStart(geneRecord[g], thisCodon)
+					delFound := codonContainsDels(thisCodon, dna.Gap) // check if codon already contains deletions
 					stop = CheckStop(geneRecord[g], thisCodon)
 					if start == true {
 						newSequence = append(newSequence, thisCodon.Seq[0])
 						newSequence = append(newSequence, thisCodon.Seq[1])
 						newSequence = append(newSequence, thisCodon.Seq[2])
 					}
+					if delFound == true { // if deletions are present in codon leave it as is (for simplicities sake; will change at a later date)
+						newSequence = append(newSequence, thisCodon.Seq[0])
+						newSequence = append(newSequence, thisCodon.Seq[1])
+						newSequence = append(newSequence, thisCodon.Seq[2])
+					}
 					if stop == true {
 						newCodon = PickStop(thisCodon)
 						newSequence = append(newSequence, newCodon.Seq[0])
 						newSequence = append(newSequence, newCodon.Seq[1])
 						newSequence = append(newSequence, newCodon.Seq[2])
 					}
-					if start == false && stop == false {
+					if start == false && stop == false && delFound == false {
```

Comment:
It seems like once a deletion happens to the sequence, it no longer goes into this if statement and therefore stops evolving with substitutions.  Is that true?

Reply:
I believe the way I've set it up here, each codon is checked for deletions and those that contain a dna.Gap are not evolved further. I had to do this because the TranslateCodons function is called during codon evolution and it cannot handle dna.Gap characters. Codons without deletions should still proceed through the function as they did previously.